sktime/forecasting/compose/_pipeline.py

```diff
@@ -1315,7 +1332,16 @@ def _fit(self, y, X, fh):
             self.forecaster_X_.fit(y=self._get_Xcols(X), fh=fh_X)
 
         self.forecaster_y_ = self.forecaster_y.clone()
-        self.forecaster_y_.fit(y=y, X=X, fh=fh)
+        if self.fit_behaviour == "use_actual":
+            self.forecaster_y_.fit(y=y, X=X, fh=fh)
+        elif self.fit_behaviour == "use_forecast":
+            if "capability:insample" not in self.forecaster_X_.get_tags():
```

Comment:
Is it checking if in sample prediction capability is present or not? Or, is it checking if the tag is present? Won't it be present always?

```suggestion
            if not self.forecaster_X_.get_tag("capability:insample"):
```

Reply:
> Won't it be present always?

Yes, the condition should e the value (True or False), not presence of the tag. This is a logic error.